src/services/lesson-convertor.js

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable no-console */
+import UserService from './user-service.js';
+import Database from '../stores/database.js';
+import DocumentService from './document-service.js';
+
+class LessonConvertor {
+  static get inject() {
+    return [Database, UserService, DocumentService];
+  }
+
+  constructor(db, userService, documentService) {
+    this.db = db;
+    this.userService = userService;
+    this.documentService = documentService;
+  }
+
+  async convertAllLessonsToDocuments() {
+    console.log('Converting lessons to documents ...');
+    const documentIds = await this.db.documentRevisions.distinct('documentId');
+    const notConvertedLessons = await this.db.lessons.find({ _id: { $nin: documentIds } }).toArray();
+
+    for (const lesson of notConvertedLessons) {
+      // eslint-disable-next-line no-await-in-loop
+      const user = await this.userService.getUserById(lesson.createdBy);
+
+      const data = {
+        roomId: lesson.roomId,
+        createdOn: lesson.createdOn,
+        createdBy: lesson.createdBy,
+        dueOn: lesson.schedule?.startsOn,
```

Comment:
To avoid `undefined`/`null` mixtures in the  we might wanna add `|| null` at the end here.

Reply:
I didn't set any fallbacks here for any of the fields, as `documentService` ensures a consistent fallback for everything in `_buildDocumentRevision`

Reply:
Ah, that's so true :man_facepalming: